opencda/scenario_testing/utils/sim_api.py

```diff
@@ -188,6 +217,21 @@ def __init__(self, scenario_params,
         weather = self.set_weather(simulation_config['weather'])
         self.world.set_weather(weather)
 
+        # Define probabilities for each type of blueprint
+        self.use_multi_class_bp = scenario_params["blueprint"][
+            'use_multi_class_bp'] if 'blueprint' in scenario_params else False
+        if self.use_multi_class_bp:
+            # bbx/blueprint meta
+            with open(scenario_params['blueprint']['bp_meta_path']) as f:
+                self.bp_meta = json.load(f)
+            self.bp_class_sample_prob = scenario_params['blueprint'][
+                'bp_class_sample_prob']
+
+            # normalize probability
+            self.bp_class_sample_prob = {
```

Comment:
good, i like the normalization here, in case the users give probabilities sum up more than 1